### ISSUE TITLE
List all architectures?

### ISSUE THREAD

**Issue by yairchu** [juhp/stack-clean-old#15]
Consider the following (real output):

```bash
> stack-clean-old list -o x86_64-osx
576M  8.6.4  (1 dirs)
8.0K  8.8.3  (2 dirs)
675M  8.10.4 (3 dirs)
 80M  8.10.7 (2 dirs)
6.4G  9.0.2  (36 dirs)
4.0K  9.2.3  (1 dirs)
3.3G  9.2.5  (28 dirs)
ghc-8.6.4
ghc-8.8.3
ghc-8.10.4
ghc-8.10.7
ghc-9.0.2
ghc-9.2.3
ghc-9.2.5

> stack-clean-old list -o aarch64-osx
 36K  9.0.2  (9 dirs)
6.0G  9.2.5  (33 dirs)
8.0G  9.2.7  (60 dirs)
469M  9.4.4  (1 dirs)
ghc-8.10.7
ghc-9.0.2
ghc-9.2.5
ghc-9.2.7
ghc-9.4.4
```

(btw not sure what the extra list of `ghc-*` at the bottom means)

Would be nicer if it would just list all architectures:

```bash
> stack-clean-old list
576M  8.6.4-x86_64-osx   (1 dirs)
8.0K  8.8.3-x86_64-osx   (2 dirs)
675M  8.10.4-x86_64-osx  (3 dirs)
 80M  8.10.7-x86_64-osx  (2 dirs)
 36K  9.0.2-aarch64-osx  (9 dirs)
6.4G  9.0.2-x86_64-osx   (36 dirs)
4.0K  9.2.3-x86_64-osx   (1 dirs)
6.0G  9.2.5-aarch64-osx  (33 dirs)
3.3G  9.2.5-x86_64-osx   (28 dirs)
8.0G  9.2.7-aarch64-osx  (60 dirs)
469M  9.4.4-aarch64-osx  (1 dirs)
```

**Comment by juhp** [comment]
The ghc-* should be referring to what is under `~/.stack/programs`.

I will try to find some time for this, but it may be a while.

**Comment by JonnyRa** [comment]
On a related note it would be nice to have some documentation for `-o` this thread helped me work it out but I didn't know what 
```
Please specify platform with --os-system (-o). 
```
meant and/or have it also be able to do all of them.

Thanks for making a useful tool though, I'm constantly struggling for diskspace at the moment and stack is one of the main culprits!  So this is very helpful :smile:

**Comment by JonnyRa** [comment]
Maybe this was confusing for me because the things it comes up with aren't obviously operating systems so I was thinking from the message it wanted something unrelated to what was presented in the list - eg like it was going to do something different based on whether I was windows or ubuntu or some other flavour of linux:

```
x86_64-linux-tinfo6-libc6-pre232
x86_64-linux-tinfo6
x86_64-linux
```

**Comment by juhp** [comment]
You are right maybe it should be renamed to `--platform` or `--os-variant` or maybe there is some more appropriate name from ghc?

The original context is #7.

**Comment by juhp** [comment]
@JonnyRa I added a little more explanation and changed the option name.

**Comment by JonnyRa** [comment]
Ah nice one, I think that will help people in the future :smile:

**Comment by juhp** [comment]
Okay, that little change is in the latest 0.4.7 release btw.

**Comment by juhp** [comment]
Okay I think I have basically implemented now: it was rather more complicated refactor.
If anyone wants to test from git?

Though the output is not quite perfect:
```shellsession
$ stack-clean-old remove 9.2.7
ghc-tinfo6-9.2.7 compiler would be removed
x86_64-linux-tinfo6:
40 dirs in ~/.stack/snapshots/x86_64-linux-tinfo6 would be removed for 9.2.7
x86_64-linux:

(use --delete (-d) for removal)
```

**Comment by juhp** [comment]
Should be better with c1814ce now.

**Comment by juhp** [comment]
Released in 0.4.8
